### PR TITLE
Add create fargateprofile support for non-eksctl-managed clusters

### LINKED ISSUE #2808: eksctl fargate support for non-eksctl clusters
**Why do you want this feature?**
I want to be able to use the eksctl fargate functionality against my cluster that wasn't created by eksctl

**What feature/behavior/change do you want?**
I want to be able to use the `eksctl create/delete fargateprofile` functionality against a cluster that was not created using eksctl.

Related issue: https://github.com/weaveworks/eksctl/issues/2174#issuecomment-719605816

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Renamed `pkg/actions/fargate` back to `pkg/fargate` as it's shared by `pkg/eks`.
Uses CFN to make the execution role.
Review by commit, otherwise the move and changes are muddled together.

Fixes #2808 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by aclevername** [comment]
Could we add a test for fargate support to the `unowned_cluster` test?

**Comment by michaelbeaumont** [comment]
Yep, in progress!

File: pkg/cfn/manager/api.go [hunk 1]
```diff
@@ -450,7 +450,7 @@ func (c *StackCollection) DescribeStacks() ([]*Stack, error) {
 	return stacks, nil
 }
 
-func (c *StackCollection) IsClusterStack() (bool, error) {
+func (c *StackCollection) HasClusterStack() (bool, error) {
```
Comment (aclevername) [inline pkg/cfn/manager/api.go:453]: snap! https://github.com/weaveworks/eksctl/pull/3073/commits/d3f5c1e5bf57a50d36b4732e0feb24720ced5228

File: integration/tests/unowned_cluster/unowned_cluster_test.go [hunk 2]
```diff
@@ -155,6 +155,43 @@ var _ = Describe("(Integration) [non-eksctl cluster & nodegroup support]", func(
...
+				WithArgs(
+					"fargateprofile",
+					"--cluster", clusterName,
+					"--name", "fp-test",
+					"--namespace", "default",
+					"--verbose", "2",
+				)
+			Expect(cmd).To(RunSuccessfullyWithOutputStringLines(
+				ContainElement(SatisfyAll(ContainSubstring("created"), ContainSubstring("fp-test"))),
+			))
```
Comment (aclevername) [inline integration/tests/unowned_cluster/unowned_cluster_test.go:169]: I could be mis-emembering, but doesn't something have to change in the way you setup a cluster for fargate to work?

Comment (michaelbeaumont) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:169]: That's why I needed to add the private subnets.

Comment (aclevername) [reply integration/tests/unowned_cluster/unowned_cluster_test.go:169]: ah my bad I was looking directly at the integration commit, missed that separate commit 😄

**Review by aclevername — APPROVED** [review]
LGTM

**Review by cPu1 — APPROVED** [review]
LGTM.

**Pull request merged**